sql/resource_sql_endpoint.go

```diff
@@ -103,6 +103,11 @@ func ResourceSqlEndpoint() common.Resource {
 			}
 			resp, err := wait.Get()
 			if err != nil {
+				// Rollback by deleting the warehouse
+				rollbackErr := w.Warehouses.DeleteById(ctx, wait.Id)
+				if rollbackErr != nil {
+					return fmt.Errorf("failed deleting warehouse: %w", rollbackErr)
```

Comment:
This will hide an original error. Maybe we can put it here as well?

Reply:
agreed, updated the error message :)